src/tests/ftest/control/config_generate_output.py

```diff
@@ -0,0 +1,447 @@
+        """Initialize a ConfigGenerateOutput object."""
+        super().__init__(*args, **kwargs)
+        # Data structure that store expected values.
+        self.nvme_socket_to_addrs = defaultdict(list)
+        self.scm_socket_to_namespaces = defaultdict(list)
+        self.pci_address_set = set()
+        self.scm_namespace_set = set()
+
+        self.interface_to_providers = defaultdict(list)
+        self.interface_set = set()
+
+    def prepare_expected_data(self):
+        """Prepare expected values.
+
+        Call dmg storage scan and network scan to collect expected values.
+        """
+        dmg = self.get_dmg_command()
+
+        # Call dmg storage scan --json.
+        storage_out = dmg.storage_scan()
+
+        # Get nvme_devices and scm_namespaces list that are buried. There's a
+        # uint64 hash of the strcut under HostStorage.
+        temp_dict = storage_out["response"]["HostStorage"]
+        struct_hash = list(temp_dict.keys())[0]
+        nvme_devices = temp_dict[struct_hash]["storage"]["nvme_devices"]
+        scm_namespaces = temp_dict[struct_hash]["storage"]["scm_namespaces"]
+
+        # Fill in the dictionary and the set for NVMe.
+        for nvme_device in nvme_devices:
+            socket_id = nvme_device["socket_id"]
+            pci_addr = nvme_device["pci_addr"]
+            self.nvme_socket_to_addrs[socket_id].append(pci_addr)
+            self.pci_address_set.add(pci_addr)
+
+        self.log.debug(
+            "nvme_socket_to_addrs = {}".format(self.nvme_socket_to_addrs))
+        self.log.info("pci_address_set = {}".format(self.pci_address_set))
+
+        # Fill in the dictionary and the set for SCM.
+        for scm_namespace in scm_namespaces:
+            numa_node = scm_namespace["numa_node"]
+            blockdev = scm_namespace["blockdev"]
+            self.scm_socket_to_namespaces[numa_node].append(blockdev)
+            self.scm_namespace_set.add(blockdev)
+
+        self.log.info(
+            "scm_socket_to_namespaces = {}".format(
+                self.scm_socket_to_namespaces))
+        self.log.info("scm_namespace_set = {}".format(self.scm_namespace_set))
+
+        # Call dmg network scan --provider=all --json.
+        network_out = dmg.network_scan(provider="all")
+
+        # Get the hash and the Interfaces list.
+        temp_dict = network_out["response"]["HostFabrics"]
+        struct_hash = list(temp_dict.keys())[0]
+        interfaces = temp_dict[struct_hash]["HostFabric"]["Interfaces"]
+
+        # Fill in the dictionary and the set for interface.
+        for interface in interfaces:
+            provider = interface["Provider"]
+            device = interface["Device"]
+            self.interface_to_providers[device].append(provider)
+            self.interface_set.add(device)
+
+        self.log.info(
+            "interface_to_providers = {}".format(self.interface_to_providers))
+        self.log.debug("interface_set = {}".format(self.interface_set))
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions